core/chains/evm/config/toml/defaults.go

```diff
@@ -65,6 +73,59 @@ func init() {
 	slices.SortFunc(DefaultIDs, func(a, b *big.Big) int {
 		return a.Cmp(b)
 	})
+
+	// read the custom defaults overrides
+	dir := env.CustomDefaults.Get()
+	if dir == "" {
+		// short-circuit; no default overrides provided
+		return
+	}
+
+	// Read directory contents
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		log.Fatalf("error reading custom defaults override directory: %v", err)
+		return
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			// Skip directories
+			continue
+		}
+
+		path := dir + "/" + entry.Name()
+		file, err := os.Open(path)
+		if err != nil {
+			log.Fatalf("error opening file (name: %v) in custom defaults override directory: %v", entry.Name(), err)
+			continue
+		}
+
+		// Read file contents
+		b, err := io.ReadAll(file)
+		if err != nil {
+			log.Fatalf("error reading file (name: %v) contents in custom defaults override directory: %v", entry.Name(), err)
+			file.Close()
+			continue
+		}
+		file.Close()
```

Comment:
Fatal will exit without executing Close:
```suggestion
    file.Close()
		if err != nil {
			log.Fatalf("error reading file (name: %v) contents in custom defaults override directory: %v", entry.Name(), err)
		}
```